lib/jake/ref.ex

```diff
@@ -1,7 +1,91 @@
 defmodule Jake.Ref do
   alias Jake.Context
 
+  def expand_ref(%Context{child: %{"$ref" => ref} = spec} = context)
+      when ref == "#" do
+    new_child = Map.drop(spec, ["$ref"]) |> Map.merge(context.root)
+    {%{context | child: new_child}, true}
+  end
+
+  def expand_ref(%Context{child: %{"$ref" => ref} = spec} = context) when is_binary(ref) do
+    uri = URI.decode(ref)
+
+    ref_map =
+      if String.starts_with?(uri, "http") do
+        process_http_path(uri)
+      else
+        process_local_path(uri) |> get_head_list_path(context.root)
+      end
+
+    new_child = Map.drop(spec, ["$ref"]) |> Map.merge(ref_map)
+
+    {%{context | child: new_child}, true}
+  end
+
   def expand_ref(%Context{child: _spec} = context) do
     {context, false}
   end
+
+  def get_head_list_path(path_list, root_schema) do
+    {head, tail} = Enum.split(path_list, -1)
```

Comment:
could we consider using https://github.com/xavier/json_pointer instead of reimplementing the same

Reply:
If I had known about this library earlier, then I certainly might have used it. However, if current implementation is ok in ref.ex and handle all cases for $ref, then we can have one less dependency in the project. What do you think?

Reply:
I would prefer to use a library that's tested by others than rolling our own, I haven't looked at the json_pointer spec, so don't know whether the current implementation is correct or not. But that's what I want to avoid if possible. Same reason why we don't roll our own json parser.